tensorflow/compiler/xla/service/gpu/ir_emitter.cc

```diff
@@ -541,7 +545,7 @@ Status IrEmitter::HandleDot(HloInstruction* dot) {
   }
 
   // Create the reduction loop which does the sum of products reduction.
-  std::unique_ptr<llvm_ir::ForLoop> reduction_loop = loop_nest.AddLoop(
+  std::shared_ptr<llvm_ir::ForLoop> reduction_loop = loop_nest.AddLoop(
```

Comment:
shared_ptr is strongly discouraged in Google code.  What is the purpose of using it here?